File: docs/vsto/codesnippet/VisualBasic/trin_excelworkbookdynamiccontrols4/ThisWorkbook.vb

Public Class ThisWorkbook

    Private Sub ThisWorkbook_Startup() Handles Me.Startup



    End Sub

    Private Sub ThisWorkbook_Shutdown() Handles Me.Shutdown

    End Sub


    '<snippet1>
    Private Sub ThisWorkbook_SheetActivate2(ByVal Sh As Object)
        Dim vstoWorksheet As Microsoft.Office.Tools.Excel.Worksheet = Nothing
        Dim interopWorksheet As Microsoft.Office.Interop.Excel.Worksheet =
            CType(Sh, Microsoft.Office.Interop.Excel.Worksheet)

        If interopWorksheet IsNot Nothing AndAlso
            Globals.Factory.HasVstoObject(interopWorksheet) Then
            vstoWorksheet = Globals.Factory.GetVstoObject(interopWorksheet)
        End If

        If vstoWorksheet IsNot Nothing Then
            ' Do something with the VSTO worksheet here. 
        End If

    End Sub
    '</snippet1> 
    Sub CreateVstoListObjectForNativeListObject()
        '<Snippet2>
        Dim vstoListObject As Microsoft.Office.Tools.Excel.ListObject = Nothing

        If Globals.Sheet1.ListObjects.Count > 0 Then
            Dim nativeListObject As Excel.ListObject = Globals.Sheet1.ListObjects(1)
            vstoListObject = Globals.Sheet1.Controls.AddListObject(nativeListObject)
        End If
        '</Snippet2>
    End Sub

    Sub CreateNamedRange()
        '<Snippet3>
        Dim range1 As Excel.Range = Globals.Sheet1.Range("A1", "D5")
        Dim namedRange1 As Microsoft.Office.Tools.Excel.NamedRange = _
            Globals.Sheet1.Controls.AddNamedRange(range1, "ChartSource")
        '</Snippet3>

        '<Snippet4>
        Globals.Sheet1.Controls.Remove("ChartSource")
        '</Snippet4>
    End Sub
    '<Snippet5>
    Sub ThisWorkbook_SheetActivate1(ByVal Sh As Object) Handles Me.SheetActivate
        Dim vstoWorksheet As Microsoft.Office.Tools.Excel.Worksheet = Nothing

        If Type.ReferenceEquals(Globals.Sheet1.InnerObject, Sh) Then
            vstoWorksheet = Globals.Sheet1.Base
        ElseIf Type.ReferenceEquals(Globals.Sheet2.InnerObject, Sh) Then
            vstoWorksheet = Globals.Sheet2.Base
        ElseIf Type.ReferenceEquals(Globals.Sheet3.InnerObject, Sh) Then
            vstoWorksheet = Globals.Sheet3.Base
        End If

        If vstoWorksheet IsNot Nothing Then
            ' Do something with the VSTO worksheet here.
        End If
    End Sub
    '</Snippet5>

End Class

File: docs/vsto/codesnippet/CSharp/trin_excelworkbookdynamiccontrols4/ThisWorkbook.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Xml.Linq;
using Microsoft.Office.Tools.Excel;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Excel = Microsoft.Office.Interop.Excel;
using Office = Microsoft.Office.Core;

namespace Trin_ExcelWorkbookDynamicControls4
{
    public partial class ThisWorkbook
    {
        private void ThisWorkbook_Startup(object sender, System.EventArgs e)
        {
        }

        private void ThisWorkbook_Shutdown(object sender, System.EventArgs e)
        {
        }

        //<snippet1>
        void ThisWorkbook_SheetActivate2(object Sh)
        {
            Microsoft.Office.Tools.Excel.Worksheet vstoWorksheet = null;
            Microsoft.Office.Interop.Excel.Worksheet interopWorksheet =
                Sh as Microsoft.Office.Interop.Excel.Worksheet;

            if (interopWorksheet != null && Globals.Factory.HasVstoObject(interopWorksheet))
            {
                vstoWorksheet = Globals.Factory.GetVstoObject(interopWorksheet);
            }

            if (vstoWorksheet != null)
            {
                // Do something with the VSTO worksheet here.
            }
        }
        //</snippet1>

        private void CreateVstoListObjectForNativeListObject()
        {
            //<Snippet2>
            Microsoft.Office.Tools.Excel.ListObject vstoListObject = null;

            if (Globals.Sheet1.ListObjects.Count > 0)
            {
                Excel.ListObject nativeListObject = Globals.Sheet1.ListObjects[1];
                vstoListObject = Globals.Sheet1.Controls.AddListObject(nativeListObject);
            }
            //</Snippet2>
        }

        private void CreateNamedRange()
        {
            //<Snippet3>
            Excel.Range range1 = Globals.Sheet1.Range["A1", "D5"];
            Microsoft.Office.Tools.Excel.NamedRange namedRange1 =
                Globals.Sheet1.Controls.AddNamedRange(range1, "ChartSource");
            //</Snippet3>

            //<Snippet4>
            Globals.Sheet1.Controls.Remove("ChartSource");
            //</Snippet4>
        }

        //<Snippet5>
        void ThisWorkbook_SheetActivate1(object Sh)
        {
            Microsoft.Office.Tools.Excel.Worksheet vstoWorksheet = null;

            if (Type.ReferenceEquals(Globals.Sheet1.InnerObject, Sh))
                vstoWorksheet = Globals.Sheet1.Base;
            else if (Type.ReferenceEquals(Globals.Sheet2.InnerObject, Sh))
                vstoWorksheet = Globals.Sheet2.Base;
            else if (Type.ReferenceEquals(Globals.Sheet3.InnerObject, Sh))
                vstoWorksheet = Globals.Sheet3.Base;

            if (vstoWorksheet != null)
            {
                // Do something with the VSTO worksheet here.
            }
        }
        //</Snippet5>

        #region VSTO Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisWorkbook_Startup);
            this.Shutdown += new System.EventHandler(ThisWorkbook_Shutdown);
        }

        #endregion

    }
}
